docs/config/README.md

```diff
@@ -1369,6 +1369,22 @@ will simply show all custom modules in the order they were defined.
 | `suffix`      | `""`                | Suffix to display immediately after the command output.                      |
 | `disabled`    | `false`             | Disables this `custom` module.                                               |
 
+#### Custom command shell
+
+`shell` accepts a non-empty list of strings, where:
+- The first string is the path to the shell to use to execute the command.
+- Other following arguments are passed to the shell.
+
+If unset, it will fallback to STARSHIP_SHELL and then to "sh" on Linux, and "cmd /C" on Windows.
+
+If `shell` is not given or only contains one element and Starship detects PowerShell will be used,
+the following arguments will automatically be added: `-NoProfile -Command -`.  
+This behavior can be avoided by explicitly passing arguments to the shell, e.g.
+
+```toml
+shell = ["pwsh", "-Command", "-"]
+```
+
```

Comment:
I've opened a PR to update the documentation for the current problem, with the code in `master`:
https://github.com/starship/starship/pull/1242

If that PR can be closed quicker, I would be in favor of using your documentation section instead and close my PR.

That said, I'd suggest we add a :::warning section about the potential terminal freeze if the shell is misconfigured, inspired by what I have in my PR. Would you be in favor to add this in your PR?

Reply:
I think your PR is a nice addition to mine, so we should keep both?

Reply:
Although there will be some conflicts.

Reply:
Yes that sounds good! I can handle the conflict, that's okay. I'm just wondering of the ordering of our PRs. We go with yours first and then we accommodate mine?

Reply:
I think so, yes. The proposed fix in the warning (using `shell = [...]`) is currently not possible because right now `shell` only accepts a string, so it would have to be slightly changed before and after merging my PR.